File: CS/ObjectClass.cs
using System;
using DevExpress.Xpo;

namespace SimpleObject {
    public class Parent : XPObject {
        public Parent(Session session) : base(session) { }
        public override void AfterConstruction() {
            base.AfterConstruction();
        }
        private string _Name;
        public string Name {
            get {
                return _Name;
            }
            set {
                SetPropertyValue("Name", ref _Name, value);
            }
        }
        [Association("ParentChildren"), Aggregated]
        public XPCollection<Child> Children {
            get { return GetCollection<Child>("Children"); }
        }
    }
    public class Child : XPObject {
        public Child(Session session) : base(session) { }
        public override void AfterConstruction() {
            base.AfterConstruction();
        }
        private string _Name;
        public string Name {
            get {
                return _Name;
            }
            set {
                SetPropertyValue("Name", ref _Name, value);
            }
        }
        private Parent _Parent;
        [Association("ParentChildren")]
        public Parent Parent {
            get {
                return _Parent;
            }
            set {
                SetPropertyValue("Parent", ref _Parent, value);
            }
        }
    }
}

File: VB/ObjectClass.vb
Imports DevExpress.Xpo

Namespace SimpleObject

    Public Class Parent
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub

        Private _Name As String

        Public Property Name As String
            Get
                Return _Name
            End Get

            Set(ByVal value As String)
                SetPropertyValue("Name", _Name, value)
            End Set
        End Property

        <Association("ParentChildren"), Aggregated>
        Public ReadOnly Property Children As XPCollection(Of Child)
            Get
                Return GetCollection(Of Child)("Children")
            End Get
        End Property
    End Class

    Public Class Child
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub

        Private _Name As String

        Public Property Name As String
            Get
                Return _Name
            End Get

            Set(ByVal value As String)
                SetPropertyValue("Name", _Name, value)
            End Set
        End Property

        Private _Parent As Parent

        <Association("ParentChildren")>
        Public Property Parent As Parent
            Get
                Return _Parent
            End Get

            Set(ByVal value As Parent)
                SetPropertyValue("Parent", _Parent, value)
            End Set
        End Property
    End Class
End Namespace
